### ISSUE TITLE
Route requestJSON is getting cached

### ISSUE THREAD

**Issue by begizi** [cypress-io/cypress#65]
The route is defined in the beforeEach with an alias and get's called several times throughout the test. When a `.wait("@aliasName")` is added after the second time the route gets called, the `its("requestJSON")` is the request from the first time the route was called.

**Comment by brian-mann** [comment]
Cannot reproduce this. My test passes:

```coffeescript
it "provides the correct requestJSON on multiple requests", ->
  post = (win, obj) ->
    win.$.ajax({
      type: "POST"
      url: "/foo"
      data: JSON.stringify(obj)
      dataType: "json"
    })

    return null

  @cy
    .server()
    .route("POST", /foo/, {}).as("getFoo")
    .window().then (win) ->
      post(win, {foo: "bar1"})
    .wait("@getFoo").its("requestJSON").should("deep.eq", {foo: "bar1"})
    .window().then (win) ->
      post(win, {foo: "bar2"})
    .wait("@getFoo").its("requestJSON").should("deep.eq", {foo: "bar2"})
```

Can you post your test code?

**Comment by brian-mann** [comment]
Couldn't ever reproduce this but XHR's have been entirely rewritten so it's naturally fixed in [`0.12.0`](https://github.com/cypress-io/cypress/wiki/changelog#0120-10232015).

**Comment by Br3d89** [comment]
Hello, I have similar issue when running same test few times (cy.wait fails). 
Here is my test_spec.js:
describe('test1', function () {
 before(function (){
    cy.server();
    cy.route('POST', 'getData', {testData: 1}).as('testData');
  })
 });

  it('Observe Creation Date', function () {
      cy.visit('/testUri/test');
      cy.wait('@testData');
  });
});

Here is my cypress.json:
{
  "baseUrl": "http://mytestSite.com",
  "viewportHeight": 1080,
  "viewportWidth": 1920,
  "chromeWebSecurity": false,
}

After clearing browser history everything works well.
Also if I run Cypress on localhost environment (baseUrl: http://localhost:8080) there is no such problem.
